### PR TITLE
SALTO-6599: Generalize group assignments for all existing Intune types

### PR DESCRIPTION
**Pull request by shoshana-houri**
_This PR is stacked. Please only review commits starting from [SALTO-6599: Generalize group assignments for all existing Intune types](https://github.com/salto-io/salto/commit/637ae472d54a54ac9152495a4220b83d40809d6e)_

Support fetch & deploy for group assignments for all existing Intune top level types.

---
See nacls structure example [here](https://github.com/salto-io/microsoft-security/tree/main/microsoft_security/Records) (all Intune top-level types).

---
_Release Notes_: 
_Microsoft Security adapter:_
* Support fetch & deploy of group assignments for all existing Intune top level types.

---
_User Notifications_: 
_Microsoft Security adapter:_
* Support fetch & deploy of group assignments for all existing Intune top level types.

### CODE REVIEW

File: packages/microsoft-security-adapter/src/definitions/deploy/intune/utils/apps_configuration.ts [hunk 1]
```diff
@@ -44,7 +47,7 @@ export const TARGET_APP_DEPLOY_DEFINITION: DeployableRequestDefinition = {
       method: 'post',
     },
     transformation: {
-      pick: ['apps'],
+      pick: [APPS_FIELD_NAME],
```
Comment (shoshana-houri) [inline packages/microsoft-security-adapter/src/definitions/deploy/intune/utils/apps_configuration.ts:50]: Just a nit, not strictly related to this PR.

File: packages/microsoft-security-adapter/src/definitions/fetch/intune/utils/group_assignments.ts
Comment (shir-reifenberg) [inline packages/microsoft-security-adapter/src/definitions/fetch/intune/utils/group_assignments.ts:1]: I think this can be in `fetch.ts`

File: packages/microsoft-security-adapter/src/definitions/deploy/intune/deploy.ts [hunk 2]
```diff
@@ -330,14 +402,14 @@ const graphBetaCustomDefinitions: DeployCustomDefinitions = {
...
-                  return {
-                    value: {
-                      deviceComplianceScheduledActionForRules: value[SCHEDULED_ACTIONS_FIELD_NAME] ?? [],
-                    },
-                  }
-                }),
+                rename: [
+                  {
+                    from: SCHEDULED_ACTIONS_FIELD_NAME,
+                    to: 'deviceComplianceScheduledActionForRules',
```
Comment (shir-reifenberg) [inline packages/microsoft-security-adapter/src/definitions/deploy/intune/deploy.ts:408]: 1. is this related to this PR? 
2. if you're changing `SCHEDULED_ACTIONS_FIELD_NAME` should we also adjust the `condition` (line 417) ? (not sure what's running first)

File: packages/microsoft-security-adapter/src/definitions/deploy/intune/utils/group_assignments.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          from: ASSIGNMENTS_FIELD_NAME,
+          to: rootField,
+          onConflict: 'skip',
+        },
+      ],
+      pick: [rootField],
+    },
+  },
+  // We can't simply use transformToCheck.pick since this request is also used for addition changes,
+  // while transformToCheck is only valid for modification changes.
```
Comment (shir-reifenberg) [inline packages/microsoft-security-adapter/src/definitions/deploy/intune/utils/group_assignments.ts:41]: IMO this comment is redundant

File: packages/microsoft-security-adapter/src/definitions/deploy/intune/utils/group_assignments.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+
+import { ASSIGNMENTS_FIELD_NAME } from '../../../../constants/intune'
+import { EndpointPath } from '../../../types'
+import { DeployableRequestDefinition } from '../../shared/types'
+import { createCustomConditionCheckChangesInFields } from '../../shared/utils'
+
+/**
+ * Creates a request to assign a resource to a group. This request is used for both addition and modification changes.
```
Comment (shir-reifenberg) [inline packages/microsoft-security-adapter/src/definitions/deploy/intune/utils/group_assignments.ts:15]: nit: I think it should be ?
```
* Creates a request to assign groups to a resource. This request is used for both addition and modification changes.
```

**Review by shir-reifenberg — APPROVED** [review]
LGTM 😎

File: packages/microsoft-security-adapter/src/definitions/deploy/intune/deploy.ts (hunk 2 above)
Comment (shoshana-houri) [reply packages/microsoft-security-adapter/src/definitions/deploy/intune/deploy.ts:408]: 1. Sorry, it's not really related. It's just a "nit" - the logic is similar but this PR just made me realize that we actually could have just use 'rename' with the right 'pick', as you originally suggested in https://github.com/salto-io/salto/pull/6506#discussion_r1744116709 :)
2. It's ok. `condition` is running first :)
